cmd/plugin_test.go

```diff
@@ -311,6 +313,179 @@ func TestStashPluginMain(t *testing.T) {
 	assert.Contains(t, output, amountDownloaded)
 }
 
+// This test addresses the issue mentioned in issue #1645. It creates a directory containing two files.
+// The paths of both files and the directory itself (without any recursive option) are added to the infile,
+// which is then passed to the plugin for upload. The test verifies the following:
+// - Both files are successfully uploaded.
+// - The directory itself is not uploaded, and no empty file with the directory name is created at the destination.
+// - An appropriate message indicating the directory upload failure is returned in the corresponding classad.
+func TestInfileUploadWithDirAndFiles(t *testing.T) {
+
+	server_utils.ResetTestState()
+	t.Cleanup(func() {
+		server_utils.ResetTestState()
+	})
+
+	if os.Getenv("RUN_STASHPLUGIN") == "1" {
+		viper.Set("Logging.Level", "debug")
+		viper.Set("TLSSkipVerify", true)
+
+		if err := config.PrintConfig(); err != nil {
+			return
+		}
+		infile := os.Getenv("TEMP_INFILE")
+		outfile := os.Getenv("TEMP_OUTFILE")
+		args := []string{"-upload", "-infile", infile, "-outfile", outfile}
+		stashPluginMain(args)
+	}
+
+	oldPrefix, err := config.SetPreferredPrefix(config.StashPrefix)
+	assert.NoError(t, err)
+
+	defer func() {
+		_, err = config.SetPreferredPrefix(oldPrefix)
+		require.NoError(t, err)
+	}()
+
+	tempUploadDir, err := os.MkdirTemp("", "TempUploadDir")
+	require.NoError(t, err)
+	defer os.RemoveAll(tempUploadDir)
+	permissions := os.FileMode(0755)
+	err = os.Chmod(tempUploadDir, permissions)
+	require.NoError(t, err)
+
+	tempObject1Content := "temp object 1 content"
+	tempObject1, err := os.Create(filepath.Join(tempUploadDir, "tempObject1"))
+	assert.NoError(t, err, "Error creating temp file")
+	_, err = tempObject1.WriteString(tempObject1Content)
+	assert.NoError(t, err, "Error writing to temp file")
+	defer tempObject1.Close()
+
+	tempObject2Content := "temp object 2 content"
+	tempObject2, err := os.Create(filepath.Join(tempUploadDir, "tempObject2"))
+	assert.NoError(t, err, "Error creating temp file")
+	_, err = tempObject2.WriteString(tempObject2Content)
+	assert.NoError(t, err, "Error writing to temp file")
+	defer tempObject2.Close()
+
+	viper.Set("Origin.EnablePublicReads", true)
```

Comment:
Move the "viper.Set" block closer to the beginning of the function. It's good practice to put the test state you are creating at the top, unless you are changing them mid way.